appserver/web/web-glue/src/main/java/com/sun/enterprise/web/PEWebContainerFeatureFactoryImpl.java

```diff
@@ -63,8 +63,8 @@ public HealthChecker getHADBHealthChecker(WebContainer webContainer) {
     }
     
     @Override
-    public ReplicationReceiver getReplicationReceiver(EmbeddedWebContainer embedded) {
-        return new PEReplicationReceiver(embedded);
+    public ReplicationReceiver getReplicationReceiver(WebContainer webContainer) {
```

Comment:
> I recognise one doesn't live without the other, but I just worry that `CatalinaWebStack` is going to end up as a bit of a monolith

It should end up as a Facade - no tomcat data manipulation should happen outside it, and therefore we know exactly what we do with it. So yes, it might grow large, but no details of plugging into Catalina -- especially modifying catalina configuration -- would be present elsewhere.

Reply:
Regarding parameter, I didn't try to change the type parameters too much, because I would otherwise be tempted to delete the entire class and all its neighbors :D